pkg/controller/config/config_controller.go

```diff
@@ -59,8 +59,7 @@ func newReconciler(mgr manager.Manager) (reconcile.Reconciler, error) {
 		return nil, err
 	}
 
-	addonsPath := filepath.Join(flag.ResourceDir, "addons")
-	addons, err := mf.NewManifest(addonsPath, flag.Recursive, mgr.GetClient())
+	addons, err := readAddons(mgr)
```

Comment:
can we pass `mgr.GetClient ` instead of `mgr`

Reply:
I need both `mgr.GetClient()` and `mgr.GetConfig()`

Reply:
the client is a dynamic client,
could you try to query CRD, like this: https://github.com/openshift/tektoncd-pipeline-operator/blob/8b7ecd3f763871596251c6191d31093adf3eb228/pkg/utils/validate/validate.go#L25